src/librustc_mir/def_use.rs

```diff
@@ -0,0 +1,170 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
```

Comment:
I've never seen the term "mention analysis" before. Google for "mention analysis data flow" brings up nothing. Where have you heard of this?